app/portainer/views/templates/templatesController.js

```diff
@@ -233,15 +238,20 @@ function ($scope, $q, $state, $transition$, $anchorScroll, ContainerService, Ima
 
     var endpointMode = $scope.applicationState.endpoint.mode;
     var apiVersion = $scope.applicationState.endpoint.apiVersion;
-
-    $q.all({
+    
+     $q.all({
       templates: TemplateService.templates(),
       volumes: VolumeService.getVolumes(),
       networks: NetworkService.networks(
         endpointMode.provider === 'DOCKER_STANDALONE' || endpointMode.provider === 'DOCKER_SWARM_MODE',
         false,
         endpointMode.provider === 'DOCKER_SWARM_MODE' && apiVersion >= 1.25
       ),
+      stacks: StackService.stacks(
+        true,
+        endpointMode.provider === 'DOCKER_SWARM_MODE' && endpointMode.role === 'MANAGER',
```

Comment:
Stack name validation should be done against all endpoints for all type of providers, you can simply avoid the provider/role check here:

```suggestion
        true,
```